### DISCUSSION TITLE
How do I debug TShock during runtime?

### DISCUSSION THREAD

**Discussion by TimGonz** [Pryaxis/TShock#1796] [Q&A] [answered]
I'm fairly new to C# so this may be a more generic/open-ended question but I'd like to be able to run TShock with a debug process attached to it. I find I learn best by seeing how this operate during runtime and I think it'd be handy to have. I'm currently using JetBrains Rider but I'll happily move to Visual Studio if need be. Thank you for your help!

**Comment by AxeelAnder** [comment, marked as answer]
You may need to change the output path of TShockAPI. 

1. Right click on the `TShockAPI` project
1. Click `Properties...`
1. Click `Debug | AnyCPU`
1. Set `Output path` to `..\TerrariaServerAPI\TerrariaServerAPI\bin\Debug\ServerPlugins\`
1. Click `OK`
1. Edit debug configurations like the picture below, it's on the left of the run button

![image](https://user-images.githubusercontent.com/25691207/82393617-8ae1de00-9a79-11ea-99f1-0a96963ee141.png)

**Attention**: If you want to push your changes, do not stage the output path change when commit

**Reply by TimGonz** [reply]
That did it, I got it working on both Visual Studio and Rider. Thanks so much! And noted about the commit warning.
